lib/orderbook/OrderBook.ts

```diff
@@ -371,6 +375,7 @@ class OrderBook extends EventEmitter {
           onUpdate && onUpdate({ type: PlaceOrderEventType.SwapSuccess, payload: swapSuccess });
         } catch (err) {
           this.logger.warn(`swap for ${portion.quantity} failed during order matching, will repeat matching routine for failed swap quantity`);
+          swapFailures.push(maker);
```

Comment:
We should add the swap reason here (and in `onUpdate` callback as well), and not to return merely the taker order. Did you plan to keep this for a different PR?

Reply:
Same question, see our discussion on discord.

Reply:
Yes I was, but I can add it to this PR in a separate commit so it completely closes #807.

Reply:
Actually I'll open a separate PR because it's not a very simple change and I think it'd be better to merge this then discuss more changes.